django_setup_configuration/config_settings.py

```diff
@@ -22,7 +23,10 @@ class ConfigField:
     field_description: str
 
 
-class ConfigSettings:
+TConfigModel = TypeVar("TConfigModel", bound=ConfigurationModel)
+
+
+class ConfigSettings(Generic[TConfigModel]):
```

Comment:
This is only for the generate documentation command right? There is probably some work to do here still to make it work, might be good to add a TODO for it or refer to it in another issue